backend/src/main/routes/user-routes.ts

```diff
@@ -9,7 +9,7 @@ import { Router } from 'express'
 export default (router: Router): void => {
   router.post('/signup', adaptRoute(makeCreateUserController()))
   router.post('/login', adaptRoute(makeLoginController()))
-  router.put('/user/:id', authorization, userController.update)
+  router.put('/user/:id', adaptMiddleware(makeAuthMiddleware()), adaptRoute(makeCreateUserController()))
```

Comment:
Está passando o createUserController na rota de update. Além disso adiciona ele no index dos controllers, e adiciona o seu import na linha 3 junto aos outros.

```suggestion
  router.put('/user/:id', adaptMiddleware(makeAuthMiddleware()), adaptRoute(makeUpdateUserController()))
```